reference/src/layout/arrays.md

```diff
@@ -0,0 +1,33 @@
+# Layout of Rust array types
+
+Array types, `[T; N]`, store `N` values of type `T` contiguously with a constant
+_stride_ (the distance between each two consecutive values).
+
+The offset of the first array element is `0`.
+
+The stride of the array is computed as the size of the element type rounded up
+to the next multiple of the alignment of the element type. That is, the _stride_
+of an array can be larger than the element size iff the alignment requirements
+of the element are larger than its size.
+
+Having a stride larger than the element size allows:
+
+```rust,ignore
+struct A(u16, u8); // size_of == 3, align_of == 4
+type B = [A; 4];  // => stride == 4 > 3
+```
+
+The size and alignment of `Vector` element types match, such that `Vector` types
+and arrays are layout compatible.
```

Comment:
Do you mean `Vec`? But the element type of a `Vec<T>` is `T`, there's no slice here.

I am not sure if `Vec` is worth mentioning here; they must be layout compatible because `Vec` allows slice indexing -- but that's a concern for whoever implements such a library.

Reply:
I meant `Vector` (as in `layout/vector.md` - the other file modified in this PR).

Reply:
Oh. The code quotes threw me off -- it's not like we have a `Vector` syntax for that in Rust.

Reply:
I guess I can hyperlink them to make that more clear.

Reply:
Done.

Reply:
Please also remove the `quotes` and mention "SIMD" -- I suppose for most Rust programmers, the first association with "vector" is `Vec`, not SIMD types.

Reply:
Done, I hope it is clearer now. Let me know if you have a better idea about how to write this.